test/aqua/operators/test_op_construction.py

```diff
@@ -395,6 +399,113 @@ def test_circuit_compose_register_independent(self):
 
         self.assertEqual(composed.num_qubits, 2)
 
+    def test_matrix_op_to_instruction(self):
+        """Test to reveal the exception that is raised when .to_instruction is called on
+        parametrized matrix op.
+        """
+        m = np.array([[0, 0, 1, 0], [0, 0, 0, -1], [1, 0, 0, 0], [0, -1, 0, 0]])
+        matrix_op = MatrixOp(m, Parameter('beta'))
+
+        # QiskitError: multiplication of Operator with ParameterExpression is not implemented
+        self.assertRaises(QiskitError, matrix_op.to_instruction)
+
+    def test_matrix_op_to_circuit(self):
+        """Test to reveal the exception that is raised when .to_circuit is called on MatrixOp
+        with parameter.
+        """
+        m = np.array([[0, 0, 1, 0], [0, 0, 0, -1], [1, 0, 0, 0], [0, -1, 0, 0]])
+        matrix_op = MatrixOp(m, Parameter('alpha'))
+
+        # QiskitError: multiplication of Operator with ParameterExpression is not implemented
+        self.assertRaises(QiskitError, matrix_op.to_circuit)
+
+    def test_primitive_op_to_matrix(self):
+        """ Test to reveal the exception that is raised when .to_matrix is called on PrimitiveOps
+        with parameter.
+        """
+        # MatrixOp
+        m = np.array([[0, 0, 1, 0], [0, 0, 0, -1], [1, 0, 0, 0], [0, -1, 0, 0]])
+        matrix_op = MatrixOp(m, Parameter('beta'))
+        # TypeError: multiplication of 'complex' and 'Parameter' is not implemented
+        self.assertRaises(TypeError, matrix_op.to_matrix)
+
+        # PauliOp
+        pauli_op = PauliOp(primitive=Pauli(label='XYZ'), coeff=Parameter('beta'))
+        # TypeError: multiplication of 'complex' and 'Parameter' is not implemented
+        self.assertRaises(TypeError, pauli_op.to_matrix)
+
+        # CircuitOp
+        qc = QuantumCircuit(2)
+        qc.cx(0, 1)
+        circuit_op = CircuitOp(qc, coeff=Parameter('alpha'))
+        # TypeError: multiplication of 'complex' and 'Parameter' is not implemented
+        self.assertRaises(TypeError, circuit_op.to_matrix)
```

Comment:
Can you wrap each `assert...` in a `with `self.subTest(msg)` context? This way we know which part is going wrong in a test.